### DISCUSSION TITLE
Is it possible to start a debug session without launching automatically a shell

### DISCUSSION THREAD

**Discussion by tpotie** [mfussenegger/nvim-dap#474] [Q&A] [answered]
Hello,

I have searched these discussions and come across some topics that seem to have scraped the surface, however, I haven't managed to get any of the parameters to work to modify the behaviour.

I am setting up `nvim-dap` with the `vscode-cpp` DAP (C++ with gdb).

When I initiate a debugging session (example with breakpoints + run), it always splits the window in 2 horizontally, with the code in the top window and the bottom has the shell (along with another buffer for REPL but which is not visible). At a later stage I will use this plugin along with `nvim-dap-ui` which makes the REPL window appear on start but also with the shell in between it and the code.

Therefore, I am trying to get this shell window sorted out before I re-enable `nvim-dap-ui`. Is it possible to set the shell to either output into the REPL or have it present in a non-visible buffer?

Has anyone got any ideas as to what I can attempt?

My config is:

```
dap.adapters.cppdbg = {
  id = 'cppdbg',
  type = 'executable',
  command = vim.fn.stdpath "data" .. '/dap/vscode-cpptools/OpenDebugAD7',
}
dap.configurations.cpp = {
  {
    name = "Launch file",
    type = "cppdbg",
    request = "launch",
    program = function()
      return vim.fn.input('Path to executable: ', vim.fn.getcwd() .. '/', 'file')
    end,
    cwd = '${workspaceFolder}',
    stopOnEntry = true,
    runInTerminal = "false",
    setupCommands = {  
        { 
            text = '-enable-pretty-printing',
            description =  'enable pretty printing',
            ignoreFailures = false 
        },
    },
  },
}
```

**Comment by mfussenegger** [comment, marked as answer]
Depends on the debug-adapter. See `:help dap-terminal`. I didn't find the `runInTerminal = false` option in the vscode-cpptools docs which you use. It looks like they don't support that?

As a workaround you could consider opening the integrated terminal in a new tab:

```lua
dap.defaults.fallback.terminal_win_cmd = 'tabnew'
```

**Reply by tpotie** [reply]
Thanks for your reply! I will use the workaround you proposed
